tests/c/udp-push-pop.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+/* This should come first. */
+#define _POSIX_C_SOURCE 200809L
+
+/*====================================================================================================================*
+ * Imports                                                                                                            *
+ *====================================================================================================================*/
+
+#include <arpa/inet.h>
+#include <assert.h>
+#include <demi/libos.h>
+#include <demi/sga.h>
+#include <demi/wait.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+
+/*====================================================================================================================*
+ * Constants                                                                                                          *
+ *====================================================================================================================*/
+
+/**
+ * @brief Data size.
+ */
+#define DATA_SIZE 64
+
+/**
+ * @brief Maximum number of iterations.
+ */
+#define MAX_ITERATIONS 1000000
+
+/*====================================================================================================================*
+ * sighandler()                                                                                                       *
+ *====================================================================================================================*/
+
+/**
+ * @brief Signal handler.
+ *
+ * @param signum Number of received signal.
+ */
+static void sighandler(int signum)
+{
+    const char *signame = strsignal(signum);
+
+    fprintf(stderr, "\nReceived %s signal\n", signame);
+    fprintf(stderr, "Exiting...\n");
+
+    exit(EXIT_SUCCESS);
```

Comment:
I'm not sure what you meant by "run inside a test harness".

Anyways, server side may not have received all packets and still it does not mean that it failed, since it's UDP.

How do you want to handle the following scenarios?
- Server received `MAX_ITERATIONS` packets
- Server received less than `MAX_ITERATIONS` packets in a given time

Reply:
Okay, we determined that a timeout from the test harness will result in the process getting a SIGTERM, which we don't have it catch, so the exit code we return from our signal handler doesn't apply to that case.